<a href="https://colab.research.google.com/github/dubalker11/gravitationalwave/blob/main/bh_rad_calculation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import astropy.units as u
import astropy.constants as ac

$R = \frac{2GM}{c^2}$

In [ ]:
bh_mass = 1 * u.solMass #solar-mass black hole
bh_rad = 2 * ac.G * bh_mass / ac.c**2
print(bh_rad.to(u.km))

2.95325007610025 km


The radius of a solar-mass black hole is 2.953 km


In [ ]:
delta_t = 0.01 * u.second
M = delta_t * ac.c**3 / (16 * np.pi * ac.G)
print(M.to(u.solMass)) #total mass of the two merging black holes

40.390629275268466 solMass


In [ ]:
h = 1e-21
delta_Earth = 1 * u.earthRad * h
print(delta_Earth.to(u.m)) #deformation of Earth

6.378099999999999e-15 m


In [ ]:
bh_r = 2 * ac.G * 30 * u.solMass / (ac.c**2)
D = bh_r / h
print(D.to(1e6 * u.parsec))

2871.249504952705 1e+06 pc


Distance of the black hole merger is 2871 1e+06 pc